llvm/lib/Transforms/Scalar/SROA.cpp

```diff
@@ -2216,86 +2317,9 @@ static VectorType *isVectorPromotionViable(Partition &P, const DataLayout &DL) {
     }
   }
 
-  // If we didn't find a vector type, nothing to do here.
-  if (CandidateTys.empty())
```

Comment:
It would be easier to review if the NFC move-to-function part were pre-committed. Not sure where the logic change is

Reply:
Ah yes, thanks for the idea!